src/server/actions/Agencies.ts

```diff
@@ -1,74 +1,258 @@
-                    { $subtract: [new Date(), '$updatedAt'] },
-                    {
-                      $multiply: ['$updateScheduleInDays', 24 * 60 * 60 * 1000],
-                    },
-                  ],
-                },
-                then: 'Completed',
-              },
-              {
-                case: {
-                  $lt: [
-                    { $subtract: [new Date(), '$updatedAt'] },
-                    {
-                      $multiply: ['$updateScheduleInDays', 24 * 60 * 60 * 1000],
-                    },
-                  ],
-                },
-                then: 'NeedsReview',
-              },
-            ],
-            default: 'Expired',
-          },
-        },
-      },
-    },
-    {
-      $addFields: {
-        daysSinceEmailSent: {
-          $cond: {
-            if: { $eq: ['$emailSentTimestamp', null] }, // Handle cases where emailSentTimestamp is not set
-            then: null,
-            else: {
-              $trunc: {
-                $divide: [
-                  {
-                    $subtract: [new Date(), '$emailSentTimestamp'],
-                  },
-                  1000 * 60 * 60 * 24, // Convert milliseconds to days
-                ],
-              },
-            },
-          },
-        },
-      },
-    },
-  ];
-  return await MongoDriver.getEntities<Agency>(AgencySchema, aggregatePipeline);
+  await dbConnect();
+  try {
+    const agencies = await AgencySchema.find({}).populate('services').exec();
+    return agencies as Agency[];
+  } catch (error) {
+    mongoErrorHandler(error as MongoError);
+    return [];
+  }
+}
+
+/**
+ *
+ * @param page Which page to get
+ * @param pageSize Number of agencies per page
+ * @returns A single page of agencies
+ * @throws ApiError if the page is less than 1
+ * @throws ApiError if the pageSize is less than 1
+ * @throws ApiError for bad mongoose request
+ */
+export async function getPaginatedAgencies(
+  page: number,
+  pageSize: number
+): Promise<Agency[]> {
+  if (page < 1 || pageSize < 1) {
+    throw new ApiError(400, errors.badRequest);
+  }
+
+  await dbConnect();
+  try {
+    const agencies = await AgencySchema.find({})
+      .populate('services')
+      .skip((page - 1) * pageSize)
+      .limit(pageSize)
+      .exec();
+    return agencies as Agency[];
+  } catch (error) {
+    mongoErrorHandler(error as MongoError);
+    return [];
+  }
+}
+
+/**
+ *
+ * @param status The status of the agencies to get (Completed, NeedsReview, Expired)
+ * @returns All agencies matching a specific status
+ * @throws ApiError if the status is not one of the three valid statuses
+ */
+export async function getAgenciesByStatus(status: string): Promise<Agency[]> {
+  if (
+    status !== 'Completed' &&
+    status !== 'NeedsReview' &&
+    status !== 'Expired'
+  ) {
+    throw new ApiError(400, errors.badRequest);
+  }
+  await dbConnect();
+  try {
+    const agencies = await AgencySchema.find({ currentStatus: status })
```

Comment:
This function isn't working since you cannot query on virtual fields as seen in their [docs](https://mongoosejs.com/docs/tutorials/virtuals.html#:~:text=documents%20with%20virtuals.-,Limitations,query%20based%20on%20Mongoose%20virtuals.&text=If%20you%20want%20to%20query,setter%20or%20pre%20save%20middleware.). We probably want to remove this server action for now since we don't necessarily need it. If we need to keep it, we could use something like shown in [this StackOverflow post](https://stackoverflow.com/questions/41636515/how-do-i-update-a-field-in-mongodb) and determine the status again like what was done in the getter for the virtual field.

Reply:
Gotcha. I'll just go ahead a remove that.